### PR TITLE
Pablo vault feat/additional functionality for instrumental

### PR DESCRIPTION
**Pull request by belousm**
## Issue
[2z4fw6z](https://app.clickup.com/t/2z4fw6z)
[2z4fww1](https://app.clickup.com/t/2z4fww1)

#### Pablo part
- *Add `get_price_of_lp_token` function to Pablo*
#### Vault part
- *Add `get_strategies ` function to Vault*

### CODE REVIEW

File: frame/composable-traits/src/vault.rs
Comment (PoisonPhang) [inline frame/composable-traits/src/vault.rs:31]: Let's go ahead and add `Eq` to this while making changes.

File: frame/pablo/src/lib.rs
Comment (PoisonPhang) [inline frame/pablo/src/lib.rs:981]: NIT:
Duplicate code can be moved to a new function

File: frame/composable-traits/src/vault.rs
Comment (belousm) [reply frame/composable-traits/src/vault.rs]: Resolved

File: frame/pablo/src/lib.rs
Comment (belousm) [reply frame/pablo/src/lib.rs]: Resolved

File: frame/composable-traits/src/vault.rs
Comment (belousm) [reply frame/composable-traits/src/vault.rs]: @PoisonPhang Because I created `get_strategies` function which use [`StrategyOverview`](https://github.com/ComposableFi/composable/blob/9f3b4c6e4c1ad70b58b4fd956c815c79eb2234a7/frame/composable-traits/src/vault.rs#L137-L139) in its annotation. So, I was needed to have this struct in the same file.

File: frame/composable-traits/src/dex.rs
Comment (saruman9) [inline frame/composable-traits/src/dex.rs:183]: Shouldn't we add this function to the `runtime` (`dali`), calculate weights for it?

P.S. The same goes for `update_accounts_deposited_one_asset_storage` function actually.

File: frame/composable-traits/src/vault.rs
Comment (saruman9) [inline frame/composable-traits/src/vault.rs:32]: I do not know how the relationship between these files is, but `StrategyOverview` appears in the following files:
- [`composable_api.rs`](https://github.com/ComposableFi/composable/blob/2805639cd55364dee41217bf68dd28b1de489aad/utils/price-feed/src/feed/composable_api.rs),
- [`composablejs` directory](https://github.com/ComposableFi/composable/tree/pablo-feat/constant_product_zapping/composablejs),
- [`integration-tests` directory](https://github.com/ComposableFi/composable/tree/pablo-feat/constant_product_zapping/integration-tests/runtime-tests),
- [`frontend` directory](https://github.com/ComposableFi/composable/tree/pablo-feat/constant_product_zapping/frontend/packages/defi-interfaces).

File: frame/composable-traits/src/vault.rs
Comment (saruman9) [reply frame/composable-traits/src/vault.rs]: Maybe my question will seem stupid, but for what reason we can't use the structure from `models.rs` file?

**Review by saruman9 — CHANGES_REQUESTED** [review]

File: frame/composable-traits/src/vault.rs
Comment (saruman9) [reply frame/composable-traits/src/vault.rs]: Maybe my question will seem stupid, but for what reason we can't use the structure from `models.rs` file?

File: frame/composable-traits/src/dex.rs
Comment (belousm) [reply frame/composable-traits/src/dex.rs:183]: To be honest, I don't know.

File: frame/composable-traits/src/vault.rs
Comment (belousm) [reply frame/composable-traits/src/vault.rs]: In the course of my work with other pallets, I noticed that the structures used in the traits are also created from the same file. Therefore, I decided to do so in order to maintain some generality.

File: frame/composable-traits/src/vault.rs
Comment (benluelo) [reply frame/composable-traits/src/vault.rs:32]: This was moved here from the vault pallet, so it already exists

File: frame/composable-traits/src/dex.rs
Comment (benluelo) [reply frame/composable-traits/src/dex.rs:183]: Only extrinsics need weights, as far as i can tell this isn't called as an extrinsic

Comment (saruman9) [reply frame/composable-traits/src/dex.rs:183]: Then be free to close (resolve) the thread if this function does not plan to be an extrinsic.

File: frame/composable-traits/src/vault.rs
Comment (belousm) [reply frame/composable-traits/src/vault.rs]: @PoisonPhang Also the `vault pallet` itself has a dependency on `composable-traits`, and then it turns out there will be a cycle if I try to import the `models.rs` file into a `composable-traits`.

File: frame/pablo/src/lib.rs
Comment (saruman9) [inline frame/pablo/src/lib.rs:979]:
```suggestion
				PoolConfiguration::StableSwap(StableSwapPoolInfo { pair, lp_token, .. }) |
				PoolConfiguration::ConstantProduct(ConstantProductPoolInfo {
					pair,
					lp_token,
					..
				}) => calculate_price_of_lp_token::<T>(pair, lp_token, &pool_account),
```
You can merge arms, please check the suggestion before accepting.

**Review by saruman9 — APPROVED** [review]
LGTM, but please find out if something needs to be changed [here](https://github.com/ComposableFi/composable/pull/1669#discussion_r972843241) due to the fact that `StrategyOverview` has changed its location before merging.

**Pull request merged**
